<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1037</width>
    <height>594</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PetNetSim editor</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>3</number>
    </property>
    <item row="1" column="0" rowspan="2">
     <widget class="Editor" name="editor"/>
    </item>
    <item row="0" column="0" colspan="2">
     <widget class="QFrame" name="top_tools_frame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,1,0,0,0,0,0,0,0">
       <property name="spacing">
        <number>3</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="mode_t_label">
         <property name="text">
          <string>Mode:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="mode_label">
         <property name="text">
          <string>Normal</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="verify_pushButton">
         <property name="text">
          <string>Verify</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="minimumSize">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="lineWidth">
          <number>2</number>
         </property>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="simulation_pushButton">
         <property name="text">
          <string>Simulation</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="simulation_run_pushButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Run</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="simulation_wait_doubleSpinBox">
         <property name="prefix">
          <string>step wait: </string>
         </property>
         <property name="suffix">
          <string> s</string>
         </property>
         <property name="maximum">
          <double>10.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.250000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="simulation_step_pushButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Step</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="simulation_reset_pushButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Reset</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="ItemProperties" name="item_properties">
      <property name="maximumSize">
       <size>
        <width>300</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <property name="currentIndex">
       <number>-1</number>
      </property>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QLabel" name="simulation_status_label">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <property name="text">
       <string>Simulation...




stats...




steps...



</string>
      </property>
     </widget>
    </item>
    <item row="3" column="0" colspan="2">
     <widget class="QLabel" name="shortcuts_help_label">
      <property name="text">
       <string>A: toggle arc mode   P: add place   T: add transition    Del: delete selected</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1037</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Editor</class>
   <extends>QGraphicsView</extends>
   <header>editor.h</header>
   <slots>
    <slot>verified_petrinet()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>ItemProperties</class>
   <extends>QStackedWidget</extends>
   <header>editor.widgets.h</header>
   <container>1</container>
   <slots>
    <slot>place_name_changed(QString)</slot>
    <slot>transition_name_changed(QString)</slot>
    <slot>place_init_tokens_changed(int)</slot>
    <slot>place_capacity_changed(int)</slot>
    <slot>transition_type_changed(QString)</slot>
    <slot>arc_inhibitor_checkbox_toggled(bool)</slot>
    <slot>arc_name_changed(QString)</slot>
    <slot>arc_n_tokens_changed(int)</slot>
    <slot>transition_timed_pdist_changed(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>verify_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>editor</receiver>
   <slot>verified_petrinet()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>530</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>411</x>
     <y>366</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulation_pushButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>simulation_editor_switched(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>621</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>733</x>
     <y>0</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulation_run_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>simulation_run()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>720</x>
     <y>36</y>
    </hint>
    <hint type="destinationlabel">
     <x>685</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulation_step_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>simulation_step()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>911</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>923</x>
     <y>-14</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>simulation_reset_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>simulation_step()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>978</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>963</x>
     <y>-5</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>simulation_editor_switched(bool)</slot>
  <slot>simulation_run()</slot>
  <slot>simulation_step()</slot>
  <slot>simulation_reset()</slot>
 </slots>
</ui>
